src/Api/Controllers/AccountsController.cs

```diff
@@ -825,5 +852,33 @@ public async Task PutUpdateTempPasswordAsync([FromBody]UpdateTempPasswordRequest
 
             throw new BadRequestException(ModelState);
         }
+
+        [HttpPost("request-otp")]
+        public async Task PostRequestOTP()
+        {
+            var user = await _userService.GetUserByPrincipalAsync(User);
+            if (user is not { UsesKeyConnector: true })
```

Comment:
It does a pattern match. We need to perform a null check, which means the alternative is `user == null || !user.UsesKeyConnector`. Rider complains that it can be replaced with a pattern match, which I personally find more readable.